presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkQueryPlanner.java

```diff
@@ -113,13 +114,15 @@ public PlanAndMore createQueryPlan(Session session, PreparedQuery preparedQuery,
         Optional<Output> output = new OutputExtractor().extractOutput(plan.getRoot());
         Optional<QueryType> queryType = getQueryType(preparedQuery.getStatement().getClass());
         List<String> columnNames = ((OutputNode) plan.getRoot()).getColumnNames();
+        PhysicalResourceSettings physicalResourceSettings = new PhysicalResourceSettings(optimizers.getPhysicalResourceOptimizer().getExecutorCount(), optimizers.getPhysicalResourceOptimizer().getHashPartitionCount());
```

Comment:
We can probably create instantiate `PhysicalResourceOptimizer` here itself instead of doing it inside `PlanOptimizers`.  It should simplify the code.

Reply:
Yes, now that the AddExchange is not updated, we are instantiating in PrestoSparkQueryPlanner.